src/lv_misc/lv_font.h

```diff
@@ -28,47 +28,93 @@ extern "C" {
  **********************/
 
-typedef struct
-{
-    uint32_t w_px : 8;
-    uint32_t glyph_index : 24;
-} lv_font_glyph_dsc_t;
+/*------------------
+ * General types
+ *-----------------*/
 
+/*Describe the properties of a glyph*/
 typedef struct
 {
-    uint32_t unicode : 21;
-    uint32_t glyph_dsc_index : 11;
-} lv_font_unicode_map_t;
-
+    uint16_t adv_w; /*The glyph needs this space. Draw the next glyph after this width. 8 bit integer, 4 bit fractional */
+    uint8_t box_w;  /*Width of the glyph's bounding box*/
+    uint8_t box_h;  /*Height of the glyph's bounding box*/
+    int8_t ofs_x;   /*x offset of the bounding box*/
+    int8_t ofs_y;  /*y offset of the bounding box*/
+    uint8_t bpp;   /*Bit-per-pixel: 1, 2, 4, 8*/
+}lv_font_glyph_dsc_t;
+
+/*Describe the properties of a font*/
 typedef struct _lv_font_struct
 {
     uint32_t unicode_first;
     uint32_t unicode_last;
-    const uint8_t * glyph_bitmap;
-    const lv_font_glyph_dsc_t * glyph_dsc;
-    const uint32_t * unicode_list;
-    const uint8_t * (*get_bitmap)(const struct _lv_font_struct *,
-                                  uint32_t); /*Get a glyph's  bitmap from a font*/
-    int16_t (*get_width)(const struct _lv_font_struct *,
-                         uint32_t);     /*Get a glyph's with with a given font*/
-    struct _lv_font_struct * next_page; /*Pointer to a font extension*/
-    uint32_t h_px : 8;
-    uint32_t bpp : 4;       /*Bit per pixel: 1, 2 or 4*/
-    uint32_t monospace : 8; /*Fix width (0: normal width)*/
-    uint16_t glyph_cnt;     /*Number of glyphs (letters) in the font*/
+
+    /*Get a glyph's  descriptor from a font*/
+    bool (*get_glyph_dsc)(const struct _lv_font_struct *, lv_font_glyph_dsc_t *, uint32_t letter);
+
+    /*Get a glyph's bitmap from a font*/
+    const uint8_t * (*get_glyph_bitmap)(const struct _lv_font_struct *, uint32_t);
+
+    /*Pointer to the font in a font pack (must have the same line height)*/
+    uint8_t line_height;      /*The real line height where any text fits*/
+    uint8_t base_line;        /*Base line measured from the top of the line_height*/
+    void * dsc;               /*Store implementation specific data here*/
+
+
+
+/*
+    - Font size:
+        not required for calculations
+
+     - Ascent
+     - Descent
+     - typoAscent
+     - typoDescent
+     - typographic descent:
+        Better to skip them to avoid confusion. Only line height and baseline matter for rendering.
+     - typoLineGap
+        Will be overwritten by the style.
+
+     - min Y  (used to quick check line intersections with other objects)
+     - max Y
+        BBoxMinXofs and BBoxMaxXofs would be useful to handle
+        if the the lines first and last character is out of the object.
+        Y is not important becasue all glyphs will fit in line_height.
+
+    - default advanceWidth
+        Not supported in text format. glyph->advacedWidth is always present
+
+    - glyphIdFormat
+        Has foxed size on text format
```

Comment:
Always with 4 bits fractional? Ok.

This feature was invented to save 4 bits when kerning not exists.

Reply:
We can't save 4 bits in text format because it is packed into a 12-byte struct.